Programming_Engine/Compute/PopulateTypes.cs

```diff
@@ -0,0 +1,205 @@
+﻿/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2020, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.oM.Programming;
+using BH.oM.Reflection.Attributes;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace BH.Engine.Programming
+{
+    public static partial class Compute
+    {
+        /***************************************************/
+        /**** Interface Methods                         ****/
+        /***************************************************/
+
+        [Description("Automatically set the DataType and default value of all parameters of the input syntax node.")]
+        [Input("node", "syntax node to modify")]
+        [Output("Modified node")]
```

Comment:
If the output is a modified node, would this not be a modify class method rather than compute?

Reply:
Yep - quite possibly. 
@adecler and I picking up a couple of other code and structural organisation for node2code and soon to be public CSharp toolkit - which I think could relate a little to best structuring of Programming_oM. Maybe. Happy to merge and mop up later if needed

Reply:
SGTM, am happy with that approach :smile: